src/plugin/range/index.js

```diff
@@ -0,0 +1,61 @@
+
+export class DayjsRange {
+  constructor(startDate, endDate) {
+    const { s, e } = startDate > endDate
+      ? { s: endDate, e: startDate }
+      : { s: startDate, e: endDate }
+
+    this.startDate = s
+    this.endDate = e
+  }
+
+  isValidRange() {
+    return this.startDate.isValid() && this.endDate.isValid()
+  }
+
+  clone() {
+    return new DayjsRange(this.startDate, this.endDate)
+  }
+
+  isOverlap(other) {
+    if (!(other instanceof DayjsRange) || !(other.isValidRange())) {
+      return false
+    }
+
+    if (
+      (this.startDate >= other.startDate && this.startDate <= other.endDate) ||
+      (this.endDate >= other.startDate && this.endDate <= other.endDate) ||
+      (this.startDate <= other.startDate && this.endDate >= other.endDate)
+    ) {
+      return true
+    }
+
+    return false
```

Comment:
Can be simplified into a single return statement

```suggestion
    return (
      (this.startDate >= other.startDate && this.startDate <= other.endDate) ||
      (this.endDate >= other.startDate && this.endDate <= other.endDate) ||
      (this.startDate <= other.startDate && this.endDate >= other.endDate)
    )
```